tests/test_pipeline_common.py

```diff
@@ -0,0 +1,256 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import copy
+import importlib
+import json
+import os
+import random
+import sys
+from pathlib import Path
+
+from transformers import PreTrainedModel, TFPreTrainedModel
+
+from .pipelines.test_pipelines_audio_classification import AudioClassificationPipelineTests
+from .pipelines.test_pipelines_feature_extraction import FeatureExtractionPipelineTests
+from .pipelines.test_pipelines_fill_mask import FillMaskPipelineTests
+from .pipelines.test_pipelines_image_classification import ImageClassificationPipelineTests
+from .pipelines.test_pipelines_text_classification import TextClassificationPipelineTests
+from .pipelines.test_pipelines_text_generation import TextGenerationPipelineTests
+
+
+pipeline_test_mapping = {
+    "audio-classification": {"test": AudioClassificationPipelineTests},
+    "feature-extraction": {"test": FeatureExtractionPipelineTests},
+    "fill-mask": {"test": FillMaskPipelineTests},
+    "image-classification": {"test": ImageClassificationPipelineTests},
+    "text-classification": {"test": TextClassificationPipelineTests},
+    "text-generation": {"test": TextGenerationPipelineTests},
+}
+
+for task, task_info in pipeline_test_mapping.items():
+    test = task_info["test"]
+    task_info["mapping"] = {
+        "pt": getattr(test, "model_mapping", None),
+        "tf": getattr(test, "tf_model_mapping", None),
+    }
+
+
+TINY_MODEL_SUMMARY_FILE_PATH = os.path.join(Path(__file__).parent.parent, "tests/utils/tiny_model_summary.json")
+with open(TINY_MODEL_SUMMARY_FILE_PATH) as fp:
+    tiny_model_summary = json.load(fp)
+
+
+PATH_TO_TRANSFORMERS = os.path.join(Path(__file__).parent.parent, "src/transformers")
+
+
+# Dynamically import the Transformers module to grab the attribute classes of the processor form their names.
+spec = importlib.util.spec_from_file_location(
+    "transformers",
+    os.path.join(PATH_TO_TRANSFORMERS, "__init__.py"),
+    submodule_search_locations=[PATH_TO_TRANSFORMERS],
+)
+transformers_module = spec.loader.load_module()
+
+
+class PipelineTesterMixin:
+    model_tester = None
+    pipieline_model_mapping = None
+    supported_frameworks = ["pt", "tf"]
+
+    @staticmethod
+    def disable_irrelevant_tests():
+        frame_f_locals = sys._getframe(1).f_locals
+        # import pdb; pdb.set_trace()
+        for task in pipeline_test_mapping.keys():
+            _task = task.replace("-", "_")
+            test_name = f"test_pipeline_{_task}"
+            if task not in frame_f_locals["pipieline_model_mapping"]:
+                frame_f_locals[test_name] = None
```

Comment:
I would advocate against relying upon this magic method here. I'd favor a `skip` over a magical non-collection of the method. Or a test that launches several tests from the available pipelines, but this nesting can be hard to test for independent tests (with the `-k` flag)

Reply:
OK, I will make it simply by just use `self.skipTest` (inside the test method).